bloom.go

```diff
@@ -124,128 +131,128 @@ func searchBloomFilterColumn(filters []BloomFilterColumn, path columnPath) Bloom
 	return nil
 }
 
-// bloomFilterEncoder is an adapter type which implements the encoding.Encoder
-// interface on top of a bloom filter.
-type bloomFilterEncoder struct {
-	filter bloom.MutableFilter
-	hash   bloom.Hash
-	keys   [128]uint64
-}
-
-func newBloomFilterEncoder(filter bloom.MutableFilter, hash bloom.Hash) *bloomFilterEncoder {
-	return &bloomFilterEncoder{filter: filter, hash: hash}
-}
+const (
+	// Size of the stack buffer used to perform bulk operations on bloom filters.
+	//
+	// This value was determined as being a good default empirically,
+	// 128 x uint64 makes a 1KiB buffer which amortizes the cost of calling
+	// methods of bloom filters while not causing too much stack growth either.
+	filterEncodeBufferSize = 128
+)
 
-func (e *bloomFilterEncoder) Bytes() []byte {
-	return e.filter.Bytes()
+type splitBlockEncoding struct {
+	encoding.NotSupported
 }
 
-func (e *bloomFilterEncoder) Reset(io.Writer) {
-	e.filter.Reset()
+func (splitBlockEncoding) EncodeBoolean(dst []byte, src []bool) ([]byte, error) {
+	splitBlockEncodeUint8(bloom.MakeSplitBlockFilter(dst), bits.BoolToBytes(src))
+	return dst, nil
 }
 
-func (e *bloomFilterEncoder) SetBitWidth(int) {
+func (splitBlockEncoding) EncodeInt32(dst []byte, src []int32) ([]byte, error) {
+	splitBlockEncodeUint32(bloom.MakeSplitBlockFilter(dst), bits.Int32ToUint32(src))
+	return dst, nil
 }
 
-func (e *bloomFilterEncoder) EncodeBoolean(data []bool) error {
-	return e.insert8(bits.BoolToBytes(data))
+func (splitBlockEncoding) EncodeInt64(dst []byte, src []int64) ([]byte, error) {
+	splitBlockEncodeUint64(bloom.MakeSplitBlockFilter(dst), bits.Int64ToUint64(src))
+	return dst, nil
 }
 
-func (e *bloomFilterEncoder) EncodeInt8(data []int8) error {
-	return e.insert8(bits.Int8ToBytes(data))
+func (e splitBlockEncoding) EncodeInt96(dst []byte, src []deprecated.Int96) ([]byte, error) {
+	splitBlockEncodeFixedLenByteArray(bloom.MakeSplitBlockFilter(dst), deprecated.Int96ToBytes(src), 12)
```

Comment:
Missed conversion of deprecated to bits here?

Reply:
The other functions above and below reference `bits` instead of `deprecated` which made me think you just missed this line while converting the API's above and below.

Reply:
Ah, that's because the `deprecated` package contains everything related to the `deprecated.Int96` type, `bits` has things for all other types.